packages/apollos-data-prayer/src/prayer-requests/data-source.js

```diff
@@ -94,26 +94,28 @@ export default class PrayerRequest extends RockApolloDataSource {
     }
   };
 
-  // QUERY PrayerRequests by Campus
-  getAllByCampus = async (campusId) => {
-    try {
-      const {
-        dataSources: { Auth },
-      } = this.context;
+  getAllByCampus = async (id = '') => {
+    const {
+      dataSources: { Auth, Campus },
+    } = this.context;
 
-      const { primaryAliasId } = await Auth.getCurrentPerson();
+    const { primaryAliasId } = await Auth.getCurrentPerson();
 
-      const prayers = await this.request('PrayerRequests/Public')
-        .filter(
-          `(CampusId eq ${
-            parseGlobalId(campusId).id
-          }) and (RequestedByPersonAliasId ne ${primaryAliasId})`
-        )
-        .get();
-      return this.sortPrayers(prayers);
-    } catch (err) {
-      throw new Error(err);
+    let campus;
+    let campusID;
+    if (id === '') {
+      campus = await Campus.getForPerson(primaryAliasId);
```

Comment:
Can you just do:
```suggestion
      campusID = await Campus.getForPerson(primaryAliasId).id;
```
and skip the campus assignment altogether?

Reply:
don't think so because `getForPerson` returns a promise, not an object

Reply:
I'm just trying to find a way to not need the `campus` call. Maybe a `.then` and pull the id off?